### PR TITLE
Pull height_child from growth monitoring forms

### PR DESCRIPTION
**Pull request by emord**
@lwyszomi I believe this will solve the issue https://manage.dimagi.com/default.asp?277399#1503311 without requiring a rebuild

### CODE REVIEW

File: custom/icds_reports/migrations/sql_templates/database_functions/insert_into_child_health_monthly.sql [hunk 1]
```diff
@@ -225,6 +225,8 @@ BEGIN
     'WHERE chm_monthly.case_id = agg.case_id AND chm_monthly.valid_in_month = 1 AND agg.month = ' || quote_literal(_start_date);
 
     EXECUTE 'UPDATE ' || quote_ident(_tablename) || ' chm_monthly SET ' ||
+      'recorded_height = CASE WHEN (date_trunc(' || quote_literal('MONTH') || ', agg.height_child_last_recorded) = ' || quote_literal(_start_date) || ') THEN agg.height_child ELSE NULL END, ' ||
```
Comment (emord) [inline custom/icds_reports/migrations/sql_templates/database_functions/insert_into_child_health_monthly.sql]: Confusingly `recorded_height` actually means "height of the child if it was recorded in this month"

File: custom/icds_reports/migrations/sql_templates/database_functions/insert_into_child_health_monthly.sql [hunk 2]
```diff
@@ -225,6 +225,8 @@ BEGIN
...
+      'height_measured_in_month = CASE WHEN (date_trunc(' || quote_literal('MONTH') || ', agg.height_child_last_recorded) = ' || quote_literal(_start_date) || ') THEN 1 ELSE 0 END, ' ||
```
Comment (jmtroth0) [inline custom/icds_reports/migrations/sql_templates/database_functions/insert_into_child_health_monthly.sql]: Why does MONTH need to be wrapped in a quote literal here?

**Review by jmtroth0 — APPROVED** [review]
bit of a nit since i don't know much about how this sql templating works

**Comment by lwyszomi** [comment]
@emord Looks good for me but, I think this will not update historical data.

**Comment by dmydlo** [comment]
Fix for the failing test in [PR](https://github.com/dimagi/commcare-hq/pull/20915).

File: custom/icds_reports/migrations/sql_templates/database_functions/insert_into_child_health_monthly.sql (hunk 2 above)
Comment (emord) [reply custom/icds_reports/migrations/sql_templates/database_functions/insert_into_child_health_monthly.sql]: I do not recall why. I've had issues with this in the past, but its likely I just did something wrong and made a bad habit of doing this

**Comment by emord** [comment]
@lwyszomi it will not fix historical data by itself. The reason this will work is that we have all growth monitoring forms in the UCR/aggregate table for the entire history. So running that SQL for all months will fix it

**Pull request merged**